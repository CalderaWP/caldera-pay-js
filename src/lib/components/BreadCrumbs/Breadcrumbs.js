// @flow
import React from 'react';

type BreadCrumbItem = {
	active: boolean,
	page: number,
	pageElementId: string,
	label: string
};

type Props = {
	onNavigate: (page:number) => void,
	items: Array<BreadCrumbItem>
}

export const BreadCrumbs = (props: Props) => {
	const {items,onNavigate} = props;
	return(
		<ol
			className="breadcrumb"

		>
			{ items.map( (item: BreadCrumbItem) => {
				const {
					active,
					page,
					pageElementId,
					label,
				} = item;

				const className = active ? 'active' : '';

				return(
					<li
						key={page}
						className={className}
					>
						<a
							onClick={() => onNavigate(page)}
							aria-controls={pageElementId}
							aria-expanded={active ? "true" : "false"}
						>
							{label}
						</a>
					</li>
				);
			})}
		</ol>
	)
};