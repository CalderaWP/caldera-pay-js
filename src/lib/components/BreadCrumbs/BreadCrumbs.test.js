import React from "react";
import renderer from "react-test-renderer";
import {BreadCrumbs} from "./Breadcrumbs";

const genericHandler = () => {};


describe( 'Breadcrumbs components', () => {

	const items = [
		{
			active: false,
			page: 1,
			pageElementId: 'page-1',
			label: 'Not Active',
			crumbName: 'hats',
			disabled: false
		},

		{
			active: true,
			page: 2,
			pageElementId: 'page-2',
			label: 'Active',
			crumbName: 'shoes',
			disabled: false
		},

	]
	it( 'shows active tab', () => {
		const component = renderer.create(
			<BreadCrumbs onNavigate={genericHandler} items={[items[1]]}/>
		);
		expect( component.toJSON() ).toMatchSnapshot();
	});

	it( 'shows not tab', () => {
		const component = renderer.create(
			<BreadCrumbs onNavigate={genericHandler} items={[items[0]]}/>
		);
		expect( component.toJSON() ).toMatchSnapshot();
	});


	it( 'Disables a tab', () => {
		const component = renderer.create(
			<BreadCrumbs onNavigate={genericHandler} items={[
				{
					active: false,
					page: 1,
					pageElementId: 'page-1',
					label: 'Should Be Disabled',
					crumbName: 'hats',
					disabled: true
				}
			]}/>
		);
		expect( component.toJSON() ).toMatchSnapshot();
	});


	it( 'works with multiple tabs', () => {
		const component = renderer.create(
			<BreadCrumbs onNavigate={genericHandler} items={items}/>
		);
		expect( component.toJSON() ).toMatchSnapshot();
	});

});