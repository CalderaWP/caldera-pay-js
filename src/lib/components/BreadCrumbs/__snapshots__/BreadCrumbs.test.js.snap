// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs components shows active tab 1`] = `
<ol
  className="breadcrumb"
>
  <li
    className="active"
  >
    <a
      aria-controls="page-2"
      aria-expanded="true"
      onClick={[Function]}
    >
      Active
    </a>
  </li>
</ol>
`;

exports[`Breadcrumbs components shows not tab 1`] = `
<ol
  className="breadcrumb"
>
  <li
    className=""
  >
    <a
      aria-controls="page-1"
      aria-expanded="false"
      onClick={[Function]}
    >
      Not Active
    </a>
  </li>
</ol>
`;

exports[`Breadcrumbs components works with multiple tabs 1`] = `
<ol
  className="breadcrumb"
>
  <li
    className=""
  >
    <a
      aria-controls="page-1"
      aria-expanded="false"
      onClick={[Function]}
    >
      Not Active
    </a>
  </li>
  <li
    className="active"
  >
    <a
      aria-controls="page-2"
      aria-expanded="true"
      onClick={[Function]}
    >
      Active
    </a>
  </li>
</ol>
`;
